### PR TITLE
more: move from getopts to clap

### LINKED ISSUE #1936: more: move from getopts to clap
Code is in:
https://github.com/uutils/coreutils/blob/master/src/uu/more/src/more.rs

Tests:
https://github.com/uutils/coreutils/blob/master/tests/by-util/test_more.rs

As example:
https://github.com/uutils/coreutils/commit/9081e120d6b866c94679d9e29c7bdb17ccccb4d0

### PR DESCRIPTION
**Pull request by k0ur0x**
Closes #1936

### CODE REVIEW

File: src/uu/more/src/more.rs [hunk 1]
```diff
@@ -138,9 +100,9 @@ fn reset_term(term: &mut redox_termios::Termios) {
     let _ = syscall::close(fd);
 }
 
-fn more(matches: getopts::Matches) {
-    let files = matches.free;
-    let mut f = File::open(files.first().unwrap()).unwrap();
+fn more(matches: ArgMatches) {
+    // FixME: currently panics but `more` should work with no arguments (ie, for piped input)
```
Comment (sylvestre) [inline src/uu/more/src/more.rs:104]: are you planning to fix that ? thanks

Comment (k0ur0x) [reply src/uu/more/src/more.rs]: I tried to but it seems like the problem doesn't have anything to do with the way we handle arguments. the `FixME` comment was actually in the code in the first place I just moved it to the right place.

File: src/uu/more/src/more.rs [hunk 2]
```diff
@@ -24,70 +23,44 @@ extern crate redox_termios;
...
+                .number_of_values(1)
+                .multiple(true),
+        )
+        .get_matches_from(args);
 
     // FixME: fail without panic for now; but `more` should work with no arguments (ie, for piped input)
-    if args.len() < 2 {
-        println!("{}: incorrect usage", args[0]);
+    if let None | Some("-") = matches.value_of(options::FILE) {
+        println!("more: incorrect usage");
```
Comment (sylvestre) [inline src/uu/more/src/more.rs:55]:
```suggestion
        println!("{}: incorrect usage", executable!());
```

**Comment by sylvestre** [comment]
Great, thanks!
I found some issues testing it. I will report them

**Comment by sylvestre** [comment]
if you are interested: 
https://github.com/uutils/coreutils/issues/1971
https://github.com/uutils/coreutils/issues/1972

**Pull request merged**